pkg/cmd/server/startconfig/start_allinone.go

```diff
@@ -0,0 +1,136 @@
+	_ "github.com/GoogleCloudPlatform/kubernetes/plugin/pkg/admission/resourcedefaults"
+	_ "github.com/GoogleCloudPlatform/kubernetes/plugin/pkg/admission/resourcequota"
+)
+
+type AllInOneOptions struct {
+	MasterArgs *MasterArgs
+	NodeArgs   *NodeArgs
+
+	WriteConfigOnly  bool
+	MasterConfigFile string
+	NodeConfigFile   string
+}
+
+const longAllInOneCommandDesc = `
+Start an OpenShift mater
+
+This command helps you launch an OpenShift server. The default mode is all-in-one, which allows
+you to run all of the components of an OpenShift system on a server with Docker. Running
+
+    $ openshift start master
+
+will start OpenShift listening on all interfaces, launch an etcd server to store persistent
+data, and launch the Kubernetes system components. The server will run in the foreground until
+you terminate the process.
+
+Note: starting OpenShift without passing the --master address will attempt to find the IP
+address that will be visible inside running Docker containers. This is not always successful,
+so if you have problems tell OpenShift what public address it will be via --master=<ip>.
+
+You may also pass an optional argument to the start command to start OpenShift in one of the
+following roles:
+
+    $ openshift start master --nodes=<host1,host2,host3,...>
+
+      Launches the server and control plane for OpenShift. You may pass a list of the node
+      hostnames you want to use, or create nodes via the REST API or 'openshift kube'.
+
+You may also pass --etcd=<address> to connect to an external etcd server instead of running an
+integrated instance, or --kubernetes=<addr> and --kubeconfig=<path> to connect to an existing
+Kubernetes cluster.
+`
+
+// NewCommandStartMaster provides a CLI handler for 'start' command
+func NewCommandStartAllInOne() *cobra.Command {
+	options := &AllInOneOptions{}
+
+	cmd := &cobra.Command{
+		Use:   "start",
+		Short: "Launch OpenShift All-In-One",
+		Long:  longAllInOneCommandDesc,
+		Run: func(c *cobra.Command, args []string) {
+			if err := options.Validate(); err != nil {
+				glog.Fatal(err)
+			}
+
+			if err := options.StartAllInOne(); err != nil {
+				glog.Fatal(err)
+			}
+		},
+	}
+
+	flags := cmd.Flags()
+
+	flags.BoolVar(&options.WriteConfigOnly, "config-only", false, "Indicates that the command should build the config that would be used to start OpenShift and do nothing else.")
+	flags.StringVar(&options.MasterConfigFile, "master-config", "", "Indicates that the command use the config file from a certain location.  If this is specified, all other options are ignored except --config-only.")
+	flags.StringVar(&options.NodeConfigFile, "node-config", "", "Indicates that the command use the config file from a certain location.  If this is specified, all other options are ignored except --config-only.")
+
+	options.MasterArgs = BindMasterArgs(flags)
+	options.NodeArgs = BindNodeArgs(flags)
+
+	bindAddrArg := BindBindAddrArg(flags)
```

Comment:
doesn't a single bind addr prevent you from setting the master, ui, and node host/ports?